haskell-src/Concordium/Crypto/BlsSignature.hs

```diff
@@ -289,6 +305,39 @@ verifyAggregate m pks sig = unsafePerformIO $ do
     withKeyArray ps [] f = withArrayLen ps f
     withKeyArray ps (pk : pks_) f = withPublicKey pk $ \pk' -> withKeyArray (pk' : ps) pks_ f
 
+-- |Verify an aggregate signature on (potentially) multiple messages by multiple keys.
+-- Each message is grouped with the set of keys that signed it.
```

Comment:
Please state here the fact that checking that there is at least one key is expected to be done by the caller now since you insisted on removing the check from the function itself.

It would not be the first time a corner-case like this was the downfall.